core/src/main/scala/akka/persistence/query/EventBySliceEnvelope.scala

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (C) 2021 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.persistence.query
+
+import java.util.Optional
+
+import akka.util.HashCode
+
+object EventBySliceEnvelope {
+  def apply[Event](
+      offset: Offset,
+      persistenceId: String,
+      sequenceNr: Long,
+      event: Event,
+      timestamp: Long,
+      entityType: String,
+      slice: Int): EventBySliceEnvelope[Event] =
+    new EventBySliceEnvelope(offset, persistenceId, sequenceNr, Option(event), timestamp, None, entityType, slice)
+
+  def create[Event](
+      offset: Offset,
+      persistenceId: String,
+      sequenceNr: Long,
+      event: Event,
+      timestamp: Long,
+      entityType: String,
+      slice: Int): EventBySliceEnvelope[Event] =
+    apply(offset, persistenceId, sequenceNr, event, timestamp, entityType, slice)
+
+  def unapply[Event](arg: EventBySliceEnvelope[Event]): Option[(Offset, String, Long, Option[Event], Long)] =
+    Some((arg.offset, arg.persistenceId, arg.sequenceNr, arg.eventOption, arg.timestamp))
+}
+
+/**
+ * Event wrapper adding meta data for the events in the result stream of
+ * [[akka.persistence.query.scaladsl.EventsBySliceQuery]] query, or similar queries.
+ *
+ * If the `event` is not defined it has not been loaded yet. It can be loaded with the
+ * [[akka.persistence.query.scaladsl.LoadEventQuery]].
+ *
+ * The `timestamp` is the time the event was stored, in milliseconds since midnight, January 1, 1970 UTC (same as
+ * `System.currentTimeMillis`).
+ */
+final class EventBySliceEnvelope[Event](
```

Comment:
replaces `akka.persistence.query.EventEnvelope` and `akka.projection.eventsourced.EventEnvelope` for eventsBySlices